lib/deploy.rb

```diff
@@ -78,38 +98,65 @@ def verify_configuration!
       log "Check done."
     end
 
-    def select_app
+    def apps_list
+      apps.map { |app| app.key.sub('/', '') }
+    end
+
+    def deployment_target
+      choice = selected_app_name(apps_list)
+      return choice unless fetch_eb
+      selected_app_name(eb_env_list(choice))
+    end
+
+    def eb_env_list(app)
+      beanstalk_application(app).environments
+    end
+
+    def selected_app_name(list)
```

Comment:
Technically this method should probably be called ``select_app_name`` as it prompts for selection, instead of remembering the selection